src/verso-manual/VersoManual/Markdown.lean

```diff
@@ -298,10 +330,67 @@ def addPartFromMarkdown {m} [Monad m]
     [MonadLiftT PartElabM m] [MonadStateOf PartElabM.State m]
     [MonadQuotation m] [AddMessageContext m] [MonadError m]
     (md : MD4Lean.Block)
-    (currentHeaderLevels : List (Nat × Nat) := [])
+    (currentHeaderLevels : HeaderMapping := [])
     (handleHeaders : List (Array Term → m Term) := [])
     (elabInlineCode : Option (Option String → String → m Term) := none)
-    (elabBlockCode : Option (Option String → Option String → String → m Term) := none) : m (List (Nat × Nat)) := do
+    (elabBlockCode : Option (Option String → Option String → String → m Term) := none) : m HeaderMapping := do
   let ctxt := {headerHandlers := ⟨handleHeaders⟩, elabInlineCode, elabBlockCode}
   let (_, { inHeaders }) ← (addPartFromMarkdownAux md |>.run ctxt |>.run {inHeaders := currentHeaderLevels})
   return inHeaders
+
+/--
+Given a finished part (which may contain subparts) render just the titles as markdown
+headings with a number of # that reflects their nesting depth. Meant to be used
+for debugging/testing only.
+-/
+def displayPartStructure (part : Verso.Doc.Elab.FinishedPart) (level : Nat := 1) : String := match part with
```

Comment:
```suggestion
open Verso.Doc.Elab in
def displayPartStructure (part : FinishedPart) (level : Nat := 1) : String := match part with
```
For a super long line like this, I think it's also nice to break the line before the `match`.